util/src/test/java/tech/pegasys/artemis/util/async/SafeFutureTest.java

```diff
@@ -309,6 +318,65 @@ public void exceptionallyCompose_shouldNotDoAnythingWhenCompletedSuccessfully()
     assertThat(result).isCompletedWithValue("Success");
   }
 
+  @SuppressWarnings("unchecked")
+  @Test
+  public void runAsync_shouldExecuteActionWithExecutorAndReturnResult() {
+    final SafeFuture<String> actionResult = new SafeFuture<>();
+    final Executor executor = mock(Executor.class);
+    final Supplier<SafeFuture<String>> action = mock(Supplier.class);
+    when(action.get()).thenReturn(actionResult);
+
+    final SafeFuture<String> result = SafeFuture.runAsync(action, executor);
+
+    final ArgumentCaptor<Runnable> captor = ArgumentCaptor.forClass(Runnable.class);
+    verify(executor).execute(captor.capture());
+    final Runnable executedRunnable = captor.getValue();
+
+    executedRunnable.run();
```

Comment:
I don' t understand why you included this line.

Reply:
Do you mind explaining it?

Reply:
Because `executor` is a mock, it won't actually run the runnable that was passed in.  So we use an `ArgumentCaptor` to capture it so we can call it ourselves when we're ready to in the test.